apps/service/src/utils/case.ts

```diff
@@ -0,0 +1,23 @@
+import { SnakeCasedPropertiesDeep, CamelCasedPropertiesDeep } from 'type-fest';
+import { toSnake, toCamel } from 'convert-keys';
+
+export type SnakeCased<T> = SnakeCasedPropertiesDeep<T>;
+export type CamelCased<T> = CamelCasedPropertiesDeep<T>;
+
+export const toSnakeCase = <T>(obj: T): SnakeCased<T> | null => {
```

Comment:
Maybe we can use this package https://www.npmjs.com/package/camelcase-keys to convert to snake case with deep mode and it is more simple.

Reply:
I think you mean: https://github.com/bendrucker/snakecase-keys

This is actually a good idea, not because the type the package provides, but because it provides a `exclude` option for excluding strings from being transformed. If everything goes well, this package should help resolve the `0xe1e` to `0_xe_1e` issue in another way. I'll give it a try.

Reply:
Sorry, https://github.com/bendrucker/snakecase-keys is right.

Reply:
Updated at 27bb01e